shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -72,100 +65,72 @@ public class ManagedBridgeController implements Reconciler<ManagedBridge>,
-        if (!status.isConditionTypeTrue(ManagedBridgeStatus.AUTHORISATION_POLICY_AVAILABLE)) {
+            // Nothing to check for Authorization Policy
+            managedBridgeService.fetchOrCreateBridgeAuthorizationPolicy(managedBridge, path);
             status.markConditionTrue(ManagedBridgeStatus.AUTHORISATION_POLICY_AVAILABLE);
-        }
 
-        NetworkResource networkResource = networkingService.fetchOrCreateBrokerNetworkIngress(managedBridge, secret, managedBridge.getSpec().getDnsConfiguration().getHost(), path);
-
-        if (!networkResource.isReady()) {
-            LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is not ready.",
-                    managedBridge.getMetadata().getName(),
-                    managedBridge.getMetadata().getNamespace());
-            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
-                status.markConditionFalse(ConditionTypeConstants.READY);
-            }
-            if (!status.isConditionTypeFalse(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE)) {
+            NetworkResource networkResource = networkingService.fetchOrCreateBrokerNetworkIngress(managedBridge, secret, managedBridge.getSpec().getDnsConfiguration().getHost(), path);
+            if (!networkResource.isReady()) {
+                LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is not ready.",
+                        managedBridge.getMetadata().getName(),
+                        managedBridge.getMetadata().getNamespace());
                 status.markConditionFalse(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE);
+                throw new ReconciliationException();
+            } else {
+                LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is ready.", managedBridge.getMetadata().getName(),
+                        managedBridge.getMetadata().getNamespace());
+                status.markConditionTrue(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE);
             }
+        } catch (ReconciliationException e) {
+            notifyManager(managedBridge);
             return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalMillis);
-        } else if (!status.isConditionTypeTrue(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE)) {
-            status.markConditionTrue(ManagedBridgeStatus.NETWORK_RESOURCE_AVAILABLE);
         }
 
-        LOGGER.info("Ingress networking resource for ManagedBridge with id '{}' in namespace '{}' is ready.", managedBridge.getMetadata().getName(), managedBridge.getMetadata().getNamespace());
-
-        if (!status.isReady()) {
+        if (managedBridgeService.isBridgeStatusChange(managedBridge)) {
+            notifyManager(managedBridge);
             metricsService.onOperationComplete(managedBridge, MetricsOperation.CONTROLLER_RESOURCE_PROVISION);
```

Comment:
@vaibhavjainwiz This is changing our metric recording logic as you record the provisioning operation as complete if there are any changes in conditions on the `ManagedBridge`. In the previous implementation, we only recorded the metric if the `ManagedBridge` had the `ready=true` condition e.g all provisioning was done.

Reply:
done

Reply:
Erm, [L#130](https://github.com/5733d9e2be6485d52ffa08870cabdee0/sandbox/blob/468203f50e91160441a47cda31801438e84d7787/shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java#L130) is an `@Override` statement.